### PR TITLE
move WHATEVER=1 to functions

### PR DESCRIPTION
**Pull request by szaimen**
It does what the title says.

Disclaimer: I didn't test the complete PR but I am pretty confident that it will work reliable this time since we already removed all problematic parts from the library.

### CODE REVIEW

**Comment by szaimen** [comment]
I think this is ready for review.

File: lib.sh [hunk 1]
```diff
@@ -1055,11 +1072,9 @@ any_key() {
 }
 
 lowest_compatible_nc() {
-if [ -z "$NC_UPDATE" ]
+if [ -z "$CURRENTVERSION" ]
```
Comment (enoch85) [inline lib.sh:1075]: We should check for `NCVERSION`

File: lib.sh [hunk 2]
```diff
@@ -1095,11 +1110,9 @@ fi
 
 # Check new version
 # shellcheck source=lib.sh
-if [ -z "$NC_UPDATE" ]
+if [ -z "$CURRENTVERSION" ]
```
Comment (enoch85) [inline lib.sh:1113]: We should check for `NCVERSION`

File: lib.sh [hunk 3]
```diff
@@ -1239,11 +1252,9 @@ printf "%b%s%b\n" "$1" "$2" "$Color_Off"
 # 2 = repository
 # Nextcloud version
 git_apply_patch() {
-if [ -z "$NC_UPDATE" ]
+if [ -z "$CURRENTVERSION" ]
```
Comment (enoch85) [inline lib.sh:1255]: We should check for `NCVERSION`

File: lib.sh (hunk 1 above)
Comment (szaimen) [reply lib.sh]: done

File: lib.sh (hunk 2 above)
Comment (szaimen) [reply lib.sh]: done

File: lib.sh (hunk 3 above)
Comment (szaimen) [reply lib.sh]: done

**Pull request merged**